src/System.Management.Automation/utils/Telemetry.cs

```diff
@@ -94,9 +95,16 @@ public static class ApplicationInsightsTelemetry
         // private const string _psCoreTelemetryKey = "ee4b2115-d347-47b0-adb6-b19c2c763808"; // Production
         private const string _psCoreTelemetryKey = "d26a5ef4-d608-452c-a6b8-a4a55935f70d"; // V7 Preview 3
 
+        // In the event there is a problem in creating the node identifier file, use the default identifier.
+        // This can happen if we are running in a system which has a read-only filesystem.
+        private static readonly Guid _defaultNodeIdentifier = new Guid("2f998828-3f4a-4741-bf50-d11c6be42f50");
+
         // Use "anonymous" as the string to return when you can't report a name
         private const string _anonymous = "anonymous";
 
+        // Use '0.0.0.0' as the string for an anonymous module version
+        private const string _anonymousVersion = "0.0.0.0";
```

Comment:
Does this need all 4 segments of the version or can we save space with just "0"?

Reply:
I thought with an actual version it might be easier on the query side ([version]::new("0") will throw), i can make this "0.0" which can still be converted to a version type to reduce payload size.